File: Java/12_BinaryTree/LC_1161_MaximumLevelSumOfABinaryTree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int maxLevelSum(TreeNode root) {
        Queue<TreeNode> queue = new LinkedList();
        queue.add(root);

        int max = Integer.MIN_VALUE;
        int level = 0;
        int max_level = 1;
        while(!queue.isEmpty()){
            int size = queue.size();
            int sum = 0;
            level++;
            for(int i = 0; i < size; i++){
                TreeNode node = queue.poll();
                sum += node.val;
                if(node.left != null){
                    queue.add(node.left);
                }

                if(node.right != null){
                    queue.add(node.right);
                }
            }

            if(sum > max){
                max_level = level;
                max = sum;
            }
        }

        return max_level;
    }
}

public class LC_1161_MaximumLevelSumOfABinaryTree {
    
}

File: Swift/Leetcode.playground/Sources/12_BinaryTree/LC_1161_MaximumLevelSumOfABinaryTree.swift
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public var val: Int
 *     public var left: TreeNode?
 *     public var right: TreeNode?
 *     public init() { self.val = 0; self.left = nil; self.right = nil; }
 *     public init(_ val: Int) { self.val = val; self.left = nil; self.right = nil; }
 *     public init(_ val: Int, _ left: TreeNode?, _ right: TreeNode?) {
 *         self.val = val
 *         self.left = left
 *         self.right = right
 *     }
 * }
 */
class Solution {
    func maxLevelSum(_ root: TreeNode?) -> Int {
        guard let root = root else {
            return 0
        }

        var max_level = 1
        var level = 0
        var max = Int.min
        var queue = [TreeNode]()
        queue.append(root)
        while !queue.isEmpty {
            level += 1

            let size = queue.count
            var sum = 0
            for _ in 0..<size {
                let node = queue.removeFirst()
                sum += node.val
                if let left = node.left {
                    queue.append(left)
                }
                if let right = node.right {
                    queue.append(right)
                }
            }

            if sum > max{
                max_level = level
                max = sum
            }
        }

        return max_level
    }
}